shared/modules/error-utils.js

```diff
@@ -0,0 +1,46 @@
+import getFirstPreferredLangCode from '../../app/scripts/lib/get-first-preferred-lang-code';
+import { setupLocale } from '../../ui';
+
+let currentLocaleMessages, enLocaleMessages;
```

Comment:
But one of the important error cases this PR is meant to handle is when attempting to intialize the UI before the background connection has been established, or if the attempt to connect to the background fails. In those cases, redux won't be instantiated yet, and we probably can't get the background state.

So it seems unavoidable that we have to directly read the messages from the file in some cases.

Maybe we can refactor this so that these messages are not left in memory?

Reply:
If the UI fails to get the initial message from the backgound, with the initial background state, we won't know the users locale. Defaulting to english or the browser preferred locale might be fine there (honestly english might be a safer default, given how often the browser preference is wrong, and given that we don't have any way to let the user change the setting on the error screen).

Once we get that initial message from the background, we have the users locale. That gets stored in Redux, but even if there is a crash before Redux is constructed, we have the locale in-memory and we can use it. It's just a matter of how to share that reference between this error code and the UI setup code. I don't think there's any reason to have two copies of the localized messages in-memory in any scenario.